Automattic/wp-calypso
# Migrate third-party API data layer handlers

This PR does a few related migrations:

The `refer` data layer handler is migrated to `dispatchRequestEx`. It records a referral to `refer.wordpress.com` when signup URL has `aff` and `cid` parameters.

The `refer` and `directly` handlers are both migrated to `registerHandlers`, so that they are moved from `build` chunk to the chunks that use them: Signup and Inline Help.

Part of #25121.

**Bundle size impact:**
It's very nice to see that code moves away from `build` into chunks that actually use it:
```
chunk                                       stat_size           parsed_size           gzip_size
async-load-blocks-inline-help-popover~help    +7314 B  (+4.1%)      +2593 B  (+2.4%)     +734 B  (+2.9%)
build                                         -8909 B  (-0.3%)      -2926 B  (-0.2%)     -984 B  (-0.3%)
manifest                                         +0 B                  +0 B               +17 B  (+0.3%)
signup                                        +1265 B  (+0.3%)       +595 B  (+0.2%)     +263 B  (+0.5%)
```

**How to test:**
The `refer` handler:
1. go to Signup and add affiliate params to the URL: `/start?aff=123&cid=456`
2. in Network Devtool, verify that a tracking request is sent to `refer.wordpress.com`

The `directly` handler:
1. go to a site where support questions from the Help sections will go to Directly, rather than Happychat or Zendesk. I had to hack the `isTicketSupportEligible` selector to always return `false`. Maybe @mattwondra has some better testing instructions for the Help pages?
2. open Inline Help and:
  i) verify that Directly was correctly initialized (there is an iframe and `window.DirectlyRTM` object)
  ii) try to post a question. You should see an "Send to our experts" label on the submit button. That means you're using Directly. I went offline before clicking that button to avoid submitting garbage questions. 

<img width="356" alt="screen shot 2018-07-31 at 16 28 20" src="https://user-images.githubusercontent.com/664258/43465968-d0265cae-94de-11e8-96ca-b8d0dcc354da.png">
